deltacat/compute/compactor/utils/system_columns.py

```diff
@@ -175,7 +175,7 @@ def project_delta_file_metadata_on_table(
 
     # append is source column
     is_source_iterator = repeat(
-        True,
+        True if delta_file_envelope.is_src_delta else False,
```

Comment:
The above comment related to `is_src_delta` also means this will no longer be boolean.

Reply:
will leave this as enhancement